integrations/opentelemetry/src/main/kotlin/com/xebia/functional/xef/opentelemetry/OpenTelemetryState.kt

```diff
@@ -34,4 +37,11 @@ class OpenTelemetryState(val tracer: Tracer) {
     Context.current().with(span)
     span.setAttribute(key, value)
   }
+
+  suspend fun setAttribute(key: String, values: List<String>) {
+    val span = currentCoroutineContext().getOpenTelemetryContext().let(Span::fromContext)
+
+    Context.current().with(span)
```

Comment:
I think I made a mistake in `setAttribute`, that was copied here:

`currentCoroutineContext().getOpenTelemetryContext()` returns the `Context` that is held in the `CoroutineContext`, so `Current.context().with(span)` seems incorrect here 🤔.

  => `Context.current()` works with `ThreadLocal`, and `currentCoroutineContext().getOpenTelemetryContext()` uses the `CoroutineContext`.

What we're doing in seems to be getting the `Span` from the `Context` in the `currentCoroutineContext()` in ` currentCoroutineContext().getOpenTelemetryContext().let(Span::fromContext)`.

And then setting that `Span` in the `Context.current()`, so the `ThreadLocal` one. There should be no need for that at all.

The goal afaik, is just to get the current `Span` from the current Context `currentCoroutineContext().getOpenTelemetryContext()`, and set an attribute to it.

So this code should be sufficient:

```kotlin
currentCoroutineContext()
  .getOpenTelemetryContext()
  .let(Span::fromContext)
  .setAttribute(AttributeKey.stringArrayKey(key), values)
```

Reply:
Good point. Yeah, I don't think we need to call `Context.current()` in the code anymore.